logicrunner/builtin/contract/member/member.go

```diff
@@ -437,25 +426,33 @@ func (m *Member) migration(txHash string, burnAddress string, amount big.Int, un
 		for _, ref := range migrationDaemonMembers {
 			migrationDaemonConfirms[ref] = false
 		}
-		dHolder := deposit.New(migrationDaemonConfirms, txHash, amount.String(), unHoldDate)
+		migrationDaemonConfirms[m.GetReference()] = true
+		pulseNum, err := foundation.GetPulseNumber()
+		if err != nil {
+			return fmt.Errorf("failed to get current pulse: %s")
+		}
+		dHolder := deposit.New(migrationDaemonConfirms, txHash, amount.String(), pulseNum)
 		txDeposit, err := dHolder.AsDelegate(tokenHolderRef)
+		fmt.Println("TTTholderRef " + tokenHolderRef.String() + " txDeposit " + txDeposit.Reference.String())
 		if err != nil {
-			return "", fmt.Errorf("failed to save as delegate: %s", err.Error())
+			return fmt.Errorf("failed to save as delegate: %s", err.Error())
 		}
 
 		err = tokenHolder.SetDeposit(txDeposit.GetReference())
 		if err != nil {
-			return "", fmt.Errorf("failed to set deposit: %s", err.Error())
+			return fmt.Errorf("failed to set deposit: %s", err.Error())
 		}
+		return nil
+	} else {
+		// Confirm transaction by migration daemon
+		fmt.Println("TTMemberRef " + m.GetReference().String() + " txHash " + txHash + "amount " + amount.String())
+		err := txDeposit.Confirm(m.GetReference(), txHash, amount.String())
```

Comment:
You should use that confirm method if we didn't found deposit. Just create new and use Confirm

Reply:
it does not work